### DISCUSSION TITLE
NavigationView

### DISCUSSION THREAD

**Discussion by wkmhv** [unoplatform/uno#17147] [General]
"msbuild-sdks": {
   "Uno.Sdk": "5.2.161"
 }

<Page x:Class="UnoApp5.Presentation.MainPage"
      xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
      xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
      xmlns:local="using:UnoApp5.Presentation"
      xmlns:uen="using:Uno.Extensions.Navigation.UI"
      xmlns:utu="using:Uno.Toolkit.UI"
      xmlns:um="using:Uno.Material"
      NavigationCacheMode="Required"
      Background="{ThemeResource BackgroundBrush}">

  <Grid utu:SafeArea.Insets="VisibleBounds">
    

    <NavigationView>
      <NavigationView.MenuItems  >
        <NavigationViewItem >
          <TabView x:Name="myTabView" Background="Aqua">
            <TabView.TabItems>
              <TabViewItem Header="GPS" x:Name="GPSTab" Tag="E_Gps" >
                <TextBox Text="Hallo GPS"></TextBox>
              </TabViewItem>
            </TabView.TabItems>
          </TabView>
        </NavigationViewItem>

      </NavigationView.MenuItems>
    </NavigationView>
 </Grid>
</Page>

![Uno5](https://github.com/unoplatform/uno/assets/50765701/b051ec94-5b28-45f5-9acc-19374a7ebd72)

What is wrong here?
It shohld look like this:
![Uno51](https://github.com/unoplatform/uno/assets/50765701/18c55455-272f-4f31-9973-be2e1194625c)

**Comment by jeromelaban** [comment]
Thanks for the report. Can you provide the [full repro](https://platform.uno/docs/articles/uno-howto-create-a-repro.html) for this issue?

**Reply by jeromelaban** [reply]
> [Uno5.zip](https://github.com/user-attachments/files/15838899/Uno5.zip)

Thanks for the repro.

Reading your XAML, you're trying to add a `TabView` inside a `NavigationViewItem`, is that the intent? I'd tend to think you want to add that to the content of the `NavigationView`, as such: http://playground.platform.uno/#9b3e6aa2

A `NavigationViewItem` is generally used to do navigation inside the central content of the `NavigationView`. 

You can find more information about the `NavigationView` control in the [Microsoft docs](https://learn.microsoft.com/en-us/windows/apps/design/controls/navigationview).

**Reply by jeromelaban** [reply]
_Make sure to answer in the same thread, it makes it easier for the community to follow, thanks!_

> Unfortunately I have to come back to the problem again.
Even though it's not the best way to use the NavigationView, I expect it to behave the same in WinUI and Uno.
This behavior only occurred with the change from the last Uno version. Until then the behavior was the same.

Thanks for the update. Even though Uno has a layout difference with WinUI, I'd like to understand why you're trying to have a `TabBar` inside this particular control. What's the use case? A `NavigationViewItem` is not supposed to be used this way, particularly because it implements selection. Having inner items that will handle selection will likely break its original behavior.

**Reply by jeromelaban** [reply]
_Please make sure to answer in the same thread, it makes it easier for the community to follow, thanks!_

> Would like something like this in NavViewWinUI.
Unfortunately this doesn't work in Uno either.

Indeed there's a problem with nested buttons in that context with Uno, most likely with the `NavigationViewItem` intercepting the clicks too early. 

Still, as I mentioned, this is not a conventional use of the `NavigationView`. You can use a `SplitView` instead, which will do what you need, or use the [hierachical navigation](https://learn.microsoft.com/en-us/windows/apps/design/controls/navigationview#hierarchical-navigation) (if you want to group individual features).

**Reply by wkmhv** [reply]
Thanks for the help.
It now also works with NavigationView.

One more question please:
How do I achieve the appearance like in WinUI throughout the application? (no thick blue rounded buttons etc.)

[Uno5.zip](https://github.com/user-attachments/files/15931554/Uno5.zip)

**Comment by wkmhv** [comment]
[Uno5.zip](https://github.com/user-attachments/files/15838899/Uno5.zip)

**Comment by wkmhv** [comment]
Thanks for the help. I'm just wondering why it looked like it did in picture two in an earlier version.

**Comment by wkmhv** [comment]
Unfortunately I have to come back to the problem again.
Even though it's not the best way to use the NavigationView, I expect it to behave the same in WinUI and Uno.
This behavior only occurred with the change from the last Uno version. Until then the behavior was the same.

[NavViewWinUi.zip](https://github.com/user-attachments/files/15859940/NavViewWinUi.zip)

[Uno5.zip](https://github.com/user-attachments/files/15859944/Uno5.zip)

**Comment by wkmhv** [comment]
Would like something like this in NavViewWinUI.
Unfortunately this doesn't work in Uno either.

[NavViewWinUi.zip](https://github.com/user-attachments/files/15902713/NavViewWinUi.zip)
[Uno5.zip](https://github.com/user-attachments/files/15902719/Uno5.zip)
